### PR TITLE
BUGFIX: SD-166 format tick value in shallow range

### PR DESCRIPTION
**Pull request by naibinh**

### CODE REVIEW

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2580/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1976/ : SUCCESS in 7m 05s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2582/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1980/ : SUCCESS in 7m 06s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1049/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2583/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/614/ : UNSTABLE in 15m 57s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1050/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2585/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/615/ : SUCCESS in 14m 37s

File: src/components/visualizations/chart/highcharts/plugins/dualAxesLabelFormatter.ts [hunk 1]
```diff
@@ -4,6 +4,41 @@ import head = require('lodash/head');
...
+ */
+function getLimitLength(value: number): number {
+    const valueStr = value.toString();
+    const isFloat = valueStr.indexOf('.') > 0;
+    const length = isFloat ? valueStr.length : valueStr.length + 1;
+    return length + DEFAULT_PADDING;
+}
+
+function isValueInShallowRange(min: number, max: number): boolean {
+    return min !== max && Math.abs(max - min) < 10;
```
Comment (ZdenekVidensky) [inline src/components/visualizations/chart/highcharts/plugins/dualAxesLabelFormatter.ts]: What is number 10? Shouldn't this be in constant?

Comment (naibinh) [reply src/components/visualizations/chart/highcharts/plugins/dualAxesLabelFormatter.ts]: Yes, it should :)

File: src/components/visualizations/chart/highcharts/plugins/dualAxesLabelFormatter.ts [hunk 2]
```diff
@@ -4,6 +4,42 @@ import head = require('lodash/head');
 import last = require('lodash/last');
 import { formatAsPercent, isInPercent } from '../customConfiguration';
 
+const DEFAULT_LIMIT_LENGTH = 4; // 0.00
```
Comment (ZdenekVidensky) [inline src/components/visualizations/chart/highcharts/plugins/dualAxesLabelFormatter.ts]: What does this comment mean?

Comment (naibinh) [reply src/components/visualizations/chart/highcharts/plugins/dualAxesLabelFormatter.ts]: It means `'0.00'.length === 4`
Does it look confusing?

Comment (naibinh) [reply src/components/visualizations/chart/highcharts/plugins/dualAxesLabelFormatter.ts]: I explained more in code comment :)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2590/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1986/ : SUCCESS in 17m 33s

**Comment by naibinh** [comment]
@ZdenekVidensky , tests have passed for both [AD](https://github.com/gooddata/gdc-analytical-designer/pull/1891) and [KD](https://github.com/gooddata/gdc-dashboards/pull/1530)

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/560/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1989/ : SUCCESS in 2m 44s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1054/ : UNSTABLE in 14m 41s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/525/ : SUCCESS in 17m 27s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2596/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1990/ : SUCCESS in 2m 05s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1055/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/2597/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/620/ : SUCCESS in 18m 30s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/561/ : SUCCESS in 21s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1991/ : SUCCESS in 13m 08s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1056/ : SUCCESS in 18m 08s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/526/ : FAILURE in 19m 59s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/562/ : SUCCESS in 24s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1992/ : SUCCESS in 2m 02s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1057/ : UNSTABLE in 17m 43s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/527/ : SUCCESS in 11m 01s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/563/ : SUCCESS in 39s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1993/ : SUCCESS in 10m 08s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/1058/ : SUCCESS in 14m 42s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/528/ : SUCCESS in 9m 18s

**Pull request merged**